pkg/build/admission/buildpodutil.go

```diff
@@ -0,0 +1,69 @@
+package admission
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+func IsBuildPod(pod *kapi.Pod) bool {
+	return hasBuildAnnotation(pod) && hasBuildEnvVar(pod)
+}
+
+func GetBuildFromPod(pod *kapi.Pod, codec runtime.Codec) (*buildapi.Build, error) {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return nil, err
+	}
+	build := &buildapi.Build{}
+	err = codec.DecodeInto([]byte(envVar.Value), build)
+	if err != nil {
+		return nil, err
+	}
+	return build, nil
+}
+
+func SetBuildInPod(build *buildapi.Build, pod *kapi.Pod, codec runtime.Codec) error {
+	envVar, err := buildEnvVar(pod)
+	if err != nil {
+		return err
+	}
+	encodedBuild, err := codec.Encode(build)
```

Comment:
> The encoded build is ephemeral in the sense that we only use it while creating the pod. Which reminds me... should this plugin/defaulter only handle create? Not sure that it makes sense to handle update.

Perhaps I misunderstood what this pull was doing: https://github.com/openshift/origin/pull/6760, but I thought it was being made for exactly this reason.  "I want to encode using a particular version because my pod only speaks v1beta3"